src/audio/drc/drc.c

```diff
@@ -140,353 +141,206 @@ static int drc_setup(struct drc_comp_data *cd, uint16_t channels, uint32_t rate)
  * End of DRC setup code. Next the standard component methods.
  */
 
-static struct comp_dev *drc_new(const struct comp_driver *drv,
-				const struct comp_ipc_config *config,
-				const void *spec)
+static int drc_init(struct processing_module *mod)
 {
-	struct comp_dev *dev = NULL;
-	struct drc_comp_data *cd = NULL;
-	const struct ipc_config_process *ipc_drc = spec;
-	size_t bs = ipc_drc->size;
+	struct module_data *md = &mod->priv;
+	struct comp_dev *dev = mod->dev;
+	struct module_config *cfg = &md->cfg;
+	struct drc_comp_data *cd;
+	size_t bs = cfg->size;
 	int ret;
 
-	comp_cl_info(&comp_drc, "drc_new()");
+	comp_info(dev, "drc_init()");
 
 	/* Check first before proceeding with dev and cd that coefficients
 	 * blob size is sane.
 	 */
 	if (bs > SOF_DRC_MAX_SIZE) {
-		comp_cl_err(&comp_drc, "drc_new(), error: configuration blob size = %u > %d",
-			    bs, SOF_DRC_MAX_SIZE);
-		return NULL;
+		comp_err(dev, "drc_init(), error: configuration blob size = %u > %d",
+			 bs, SOF_DRC_MAX_SIZE);
+		return -EINVAL;
 	}
 
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-	dev->ipc_config = *config;
-
 	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
 	if (!cd)
-		goto fail;
-
-	comp_set_drvdata(dev, cd);
+		return -ENOMEM;
 
+	md->private = cd;
 	cd->drc_func = NULL;
```

Comment:
you can also delete this line since `rzalloc` makes this null anyways

Reply:
Yep, fixed.